avocado/core/test.py

```diff
@@ -383,6 +384,11 @@ def _start_logging(self):
                                                        formatter,
                                                        self._ssh_logfile)
 
+        if isinstance(sys.stdout, output.LoggingFile):
+            sys.stdout.add_logger(logging.getLogger("avocado.test.stdout"))
+        if isinstance(sys.stderr, output.LoggingFile):
+            sys.stderr.add_logger(logging.getLogger("avocado.test.stderr"))
```

Comment:
We couldn't reproduce it on our systems, because we have `python-yum` installed. The `import yum` disables the `logging.raiseExceptions` which makes the logging error message disappear.

The reproducer is:

```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index a1c7c38..dae5c70 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -146,6 +146,9 @@ class Test(unittest.TestCase):
         :param job: The job that this test is part of.
         :raises: :class:`avocado.core.test.NameNotTestNameError`
         """
+        logging.raiseExceptions = True
+        a = logging.getLogger("avocado.test.stderr")
+        a.critical("asdf")
         def record_and_warn(*args, **kwargs):
             """ Record call to this function and log warning """
             if not self.__log_warn_used:
```

and it could be even trickier, when the user defines custom handler:

```diff
diff --git a/avocado/core/test.py b/avocado/core/test.py
index a1c7c38..588fb03 100644
--- a/avocado/core/test.py
+++ b/avocado/core/test.py
@@ -146,6 +146,10 @@ class Test(unittest.TestCase):
         :param job: The job that this test is part of.
         :raises: :class:`avocado.core.test.NameNotTestNameError`
         """
+        logging.raiseExceptions = True
+        a = logging.getLogger("foo")
+        a.propagate = False
+        a.critical("asdf")
         def record_and_warn(*args, **kwargs):
             """ Record call to this function and log warning """
             if not self.__log_warn_used:
```

I'll take a look at how to avoid this and whether it could be triggered by something else as well.

PS: The later is also an issue as the user might override the `Test.__init__`, or `Test._run_avocado`.